flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/dialect/psql/PostgresTypeMapper.java

```diff
@@ -49,6 +49,8 @@
     // float <=> float8
     // boolean <=> bool
     // decimal <=> numeric
+    private static final String PG_SMALLSERIAL = "smallserial";
```

Comment:
WDYT about adding new test cases for the two new types?

Reply:
In fact a test for `smallserial` is already present at https://github.com/apache/flink/blob/99c0438253efaf0a9626a807ce2b882402a367d1/flink-connectors/flink-connector-jdbc/src/test/java/org/apache/flink/connector/jdbc/catalog/PostgresCatalogTestBase.java#L360

before pgjdbc 42.2.17 `smallserial` was erroneously handled as `int2` and test passed. In 42.2.17 it was fixed and this is the reason it starts failing.
Probably it makes sense to add test for `_smallserial`

Reply:
done